main.cpp

```diff
@@ -14,15 +15,18 @@ int isNumeric(const string & chars) {
 }
 
 int main (int argc, char *argv[]) {
+    
     Entity ent1(150, 10, "Maple");
     Entity ent2(45, 30, "Sally");
-    if(argc != 7 || !(isNumeric(argv[2]) && isNumeric(argv[3]) && isNumeric(argv[5]) && isNumeric(argv[6])) ){
+    if(argc != 3){
         std::cout << "Wrong initial data given, using default values." << std::endl;
     }else{
-        Entity ent1FromArgs(stof(argv[2]), stof(argv[3]), argv[1]);
-        Entity ent2FromArgs(stof(argv[5]), stof(argv[6]), argv[4]);
-        ent1 = ent1FromArgs;
-        ent2 = ent2FromArgs;
+        try{
+            ent1 = Entity::parseUnit(argv[1]);
+            ent2 = Entity::parseUnit(argv[2]);
+        }catch(std::exception& e){
+            std::cout << "An error occured while the program was running: " << e.what() << std::endl;
+        }
```

Comment:
Ha elhasal a dolog, akar igy, akar ugy, akkor nem kellene 0-val visszaterni. Illetve azt jol ertem, hogy akkor is csinalsz Gam-et, ha elszallt? Azt nem kellene.